### PR TITLE
[386] Set up Locations

### LINKED ISSUE #386: Introduce missing concept: Location
In meeting with our stakeholders we realized that there was a concept that we were missing that lie between a Facility and an Enclosure(Tank). 

Facilities may (or may not) have_many locations within that given facility where Enclosures, Cohort's, and Animals are kept. We will want our users to know _where_ in a facility a given Cohort or Echlosure is located. 

We should likely break up the relationship of Enclosure to Facility, and instead let an Enclosure belong to a Facility _through_ a Location.

### PR DESCRIPTION
**Pull request by bkrigel**
# Checklist:

- [x] I have performed a self-review of my own code,
- [x] I have commented my code, particularly in hard-to-understand areas,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works,
- [x] New and existing unit tests pass locally with my changes ("bundle exec rake"),
- [x] Title include "WIP" if work is in progress.

Resolves #386

### Description
Introduces the `Location` model to the application, associates Location to Facility and Enclosure, write related feature specs

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?
Feature specs to test the index, show, new, update, and destroy actions

### Screenshots
Screen Shot 2020-09-24 at 1 52 01 PM

### CODE REVIEW

File: spec/models/location_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Location, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (benreyn) [inline spec/models/location_spec.rb:5]: Are there any tests you think would be valuable to add here? if not, should we just delete this file until we do?

File: spec/fixtures/enclosures.csv [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-AOP,Enclosure Number 2
-AOP,Enclosure Number 2
-QQQ,Enclosure Number 1
-,
\ No newline at end of file
+facility_code,location_name,name
+AOP,SeaLab2020,Enclosure Number 1
+AOP,Atlantis,Enclosure Number 2
+AOP,SeaWorld,Enclosure Number 2
+QQQ,OceanPlace,Enclosure Number 1
```
Comment (benreyn) [inline spec/fixtures/enclosures.csv:5]: 🤣

**Review by benreyn — APPROVED** [review]
There are ALOT of lines of code here so I only really skimmed this PR, but everything seems to be in order. @CraigJZ, would you mind doing a second pass?

File: spec/models/location_spec.rb (hunk 1 above)
Comment (smkopp92) [reply spec/models/location_spec.rb]: Good catch, thanks Ben!

**Pull request merged**